torch/nn/modules/loss.py

```diff
@@ -842,42 +842,47 @@ def forward(self, input1, input2, target):
         return F.margin_ranking_loss(input1, input2, target, self.margin, self.size_average)
 
 
-class MultiMarginLoss(Module):
+class MultiMarginLoss(_WeightedLoss):
     r"""Creates a criterion that optimizes a multi-class classification hinge
     loss (margin-based loss) between input `x` (a 2D mini-batch `Tensor`) and
     output `y` (which is a 1D tensor of target class indices,
     `0` <= `y` <= `x.size(1)`):
 
     For each mini-batch sample::
 
-        loss(x, y) = sum_i(max(0, (margin - x[y] + x[i]))^p) / x.size(0)
+        loss(x, y) = sum_i(max(0, w[y] * (margin - x[y] + x[i]))^p) / x.size(0)
```

Comment:
Oh this ugly expression... It's not your fault so I'm not asking you to fix this, but we should change it to using `.. math::` or something.